File: sample-servers/authorization-server/src/main/java/com/dm/server/authorization/dto/ClientDto.java
package com.dm.server.authorization.dto;

import com.dm.common.dto.AuditableDto;
import com.dm.common.dto.IdentifiableDto;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Set;

/**
 * <p>ClientDto</p>
 *
 * @author ldwqh0@outlook.com
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class ClientDto extends AuditableDto implements IdentifiableDto<String> {

    private static final long serialVersionUID = 2506603023979985011L;

    private String id;

    private String name;

    private String secret;

    private Set<String> authorizedGrantTypes;

    private Set<String> scope;

    private Integer accessTokenValiditySeconds;

    private Integer refreshTokenValiditySeconds;
}

File: sample-servers/authorization-server/src/main/kotlin/com/dm/server/authorization/dto/ClientDto.kt
package com.dm.server.authorization.dto

import com.dm.server.authorization.entity.Client

data class ClientDto(
    val id: String? = null,

    val name: String = "",

    val secret: String = "",

    val type: Client.Type = Client.Type.CLIENT,

    val accessTokenValiditySeconds: Int = 3600,

    val refreshTokenValiditySeconds: Int = 7200,

    val requirePkce: Boolean = false,

    val autoApprove: Boolean = false,

    val authorizedGrantTypes: Set<String> = setOf(),

    val scopes: Set<String> = setOf(),

    val registeredRedirectUris: Set<String> = setOf()
)
